### PR TITLE
Adds docs for RPC workflows to docs site

### PR DESCRIPTION
**Pull request by cgranleese-r7**
Adds documentation for RPC. Adds information on setting up the server and client, as well as some example commands and workflows. 

Also reshuffled some `nav_orders` within `docs/navigation.rb` to make more sense in terms of grouping.

## Verification

List the steps needed to make sure this thing works

- [ ] Follow this [documentation](https://github.com/rapid7/metasploit-framework/tree/3beb3d775f7a9166dadfbbd9e392b443c51a7afd/docs#production-build) to test locally.

### CODE REVIEW

File: docs/navigation.rb [hunk 1]
```diff
@@ -97,20 +97,28 @@ def without_prefix(prefix)
...
+            path: 'How-to-use-a-reverse-shell-in-Metasploit.md',
+            nav_order: 6
+          },
+          {
+            path: 'How-to-use-msfvenom.md',
+            nav_order: 7
           },
           {
-            path: 'How-to-use-a-reverse-shell-in-Metasploit.md'
+            path: 'How-to-use-Metasploit-RPC.md',
```
Comment (adfoster-r7) [inline docs/navigation.rb:120]: I'd maybe say this should be in advanced 👀

Comment (cgranleese-r7) [reply docs/navigation.rb:120]: Sure 👍

File: docs/metasploit-framework.wiki/How-to-use-Metasploit-RPC.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
+# RPC API
```
[the remark this answers is not in the document]
Comment (cgranleese-r7) [reply docs/metasploit-framework.wiki/How-to-use-Metasploit-RPC.md:1]: Yea, looking at this I'm also not sure 😄

[the remark this answers is not in the document]
Comment (cgranleese-r7) [reply docs/metasploit-framework.wiki/How-to-use-Metasploit-RPC.md:1]: I'll convert this to a draft until I get the RPC JSON API documentation wrote up and added to this PR 👍

**Comment by cgranleese-r7** [comment]
@msjenkins-r7 retest this please

**Comment by adfoster-r7** [comment]
## Release Notes

Adds documentation for both the JSON and MessagePack Metasploit RPC APIs - which is useful for programatically interacting with Metasploit

**Pull request merged**